openedx/core/lib/xblock_serializer/block_serializer.py

```diff
@@ -112,6 +112,9 @@ def _serialize_html_block(self, block) -> etree.Element:
             olx_node.attrib["editor"] = block.editor
         if block.use_latex_compiler:
             olx_node.attrib["use_latex_compiler"] = "true"
+
+        # Escape any CDATA special chars
+        block.data = block.data.replace("]]>", "]]&gt;")
         olx_node.text = etree.CDATA("\n" + block.data + "\n")
```

Comment:
I totally thought that this `etree.CDATA` function would do the escaping for us. I guess not!